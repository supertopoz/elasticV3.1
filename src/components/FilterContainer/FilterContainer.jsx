import React from 'react';
import styled from 'styled-components'
import PropTypes from "prop-types";
import { MiniCard } from "../MiniCard/MiniCard";

const FilterBlock = styled('div')`
  background: white;
  min-height: 50px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
  color:white;
  font-size: 2em;
  display: grid;
  grid-template-rows: 1fr 3fr;
   @media (max-width: 499px) {grid-template-rows: 1fr 5fr;}
`

const FilterBlockHeader = styled('div')`
  color: black;
  padding: 0.5rem;
  font-size: 1rem;
`

const FilterBlockBody = styled('div')`
  overflow: auto;
  display: grid;
  grid-gap: 10px;
  padding: 15px;
  grid-template-columns: repeat(7, 1fr);
  @media (max-width: 499px) {grid-template-columns: repeat(3, 1fr);}
  @media (min-width: 500px) {}
  @media (min-width: 700px) {}
 
`
export const FilterContainer = (props) => {
	return (
		<FilterBlock>
		  <FilterBlockHeader>FILTER CLUSTERS</FilterBlockHeader>
      <FilterBlockBody>
      { props.clusterErrors.map((item, index) => 
        <MiniCard key={`card- ${index}`} cardData={item} listFunc={props.filter} />
        )}
      </FilterBlockBody>
		</FilterBlock>
	)
}

FilterContainer.propTypes = {
  clusterErrors: PropTypes.array
};