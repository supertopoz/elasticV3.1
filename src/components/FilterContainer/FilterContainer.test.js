import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import { FilterContainer } from './FilterContainer';

describe('Header', () => {
  test('is rendered', () => {
    const component = render(
      <FilterContainer />
    );

    expect(component)
      .toMatchSnapshot();
  });
});