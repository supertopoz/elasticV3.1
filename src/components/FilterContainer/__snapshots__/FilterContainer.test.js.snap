// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is rendered 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bxivhb gxsIFa",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-ifAKCX hYJTjV",
        },
        "children": Array [
          Object {
            "data": "FILTER CLUSTERS",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-EHOje bibiRa",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "sc-bdVaJa UcwUE",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "sc-bwzfXH huUtDP",
                  },
                  "children": Array [
                    Object {
                      "data": "name",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "sc-htpNat jhPcgS",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "sc-htpNat jhPcgS",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "sc-bwzfXH huUtDP",
                    },
                    "children": Array [
                      Object {
                        "data": "name",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "sc-EHOje bibiRa",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "sc-bdVaJa UcwUE",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "sc-bwzfXH huUtDP",
                },
                "children": Array [
                  Object {
                    "data": "name",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "sc-htpNat jhPcgS",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "sc-htpNat jhPcgS",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "sc-bwzfXH huUtDP",
                  },
                  "children": Array [
                    Object {
                      "data": "name",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "sc-ifAKCX hYJTjV",
          },
          "children": Array [
            Object {
              "data": "FILTER CLUSTERS",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
